cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -501,8 +501,13 @@ where
 			.runtime_api()
 			.extract_fraud_proofs(&BlockId::Hash(primary_hash), extrinsics)?;
 
-		for _fraud_proof in fraud_proofs {
-			// TODO: Remove the corresponding invalid receipt entry from cache.
+		for fraud_proof in fraud_proofs {
+			let bad_receipt_number = fraud_proof.parent_number + 1;
+			crate::aux_schema::delete_bad_receipt(
+				&*self.client,
+				bad_receipt_number,
+				fraud_proof.bad_signed_receipt_hash,
+			)?;
```

Comment:
Are we potentially deleting something we've just inserted a few lines above? If so then it feels like it would be much more efficient to accumulate changes before actually doing something with the database.

Also is it guaranteed that we delete something that is actually stored? In other words is it possible that we're deleting something that we didn't have in the first place (we haven't seen ER corresponding to FP), in which case the first error returned from `delete_bad_receipt` will prevent others from being deleted, leaking storage items forever.